### PR TITLE
 Sync Sequential.fit() with Model.fit()

### PR DESCRIPTION
**Pull request by ozabluda**
... and improve docstrings.

A couple of API changes, all backward compatible (additions and more permissive), except that now Model.fit(epoch=1) is the default (matching Sequential.fit(). I didn't find any example/*.py affected by this.

### CODE REVIEW

File: tests/keras/test_callbacks.py [hunk 1]
```diff
@@ -247,12 +247,12 @@ def test_EarlyStopping_reuse():
     stopper = callbacks.EarlyStopping(monitor='acc', patience=patience)
     weights = model.get_weights()
 
-    hist = model.fit(data, labels, callbacks=[stopper])
+    hist = model.fit(data, labels, callbacks=[stopper], epochs=20)
```
Comment (fchollet) [inline tests/keras/test_callbacks.py]: This would slow down the tests and should be set to a lower value (e.g. 10). Is this change related to the rest of the PR? Doesn't look like it is...

Comment (ozabluda) [reply tests/keras/test_callbacks.py]: Without this change (introduced in the second commit) tests fail (see first commit). Maybe this change is just papering over a real bug?

I chose epochs=20, because earlier in the same function epochs=20 is used. In the whole file, epochs=20 is used 8 times, and epochs=10 is used 0 times.

Comment (fchollet) [reply tests/keras/test_callbacks.py]: Ok then.

**Comment by ozabluda** [comment]
Put docstrings into a different PR https://github.com/fchollet/keras/pull/8233

**Comment by ozabluda** [comment]
@fchollet, maybe this PR and https://github.com/fchollet/keras/pull/8233 are good candidates for the upcoming Keras 1.1.0, since they are supposedly API and docs improvements.

**Comment by fchollet** [comment]
@ozabluda I will take a look, thanks.
